tests/unittests/tests-mtd_flashpage/tests-mtd_flashpage.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2016 OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ */
+#include <string.h>
+#include <errno.h>
+
+#include "embUnit.h"
+
+#include "mtd.h"
+#include "mtd_flashpage.h"
+
+static mtd_dev_t _dev = MTD_FLASHPAGE_INIT_VAL(8);
+static mtd_dev_t *dev = &_dev;
+
+static void setup_teardown(void)
+{
+    mtd_erase(dev, (FLASHPAGE_NUMOF - 1) * FLASHPAGE_SIZE, dev->pages_per_sector * dev->page_size);
```

Comment:
I think the problem is here, this doesn't take into account `CPU_FLASH_BASE`, thus it tries to erase at an address with offset 0. This problem is here and below.

Reply:
I think it should be `CPU_FLASH_BASE + ((FLASHPAGE_NUMOF - 1) * FLASHPAGE_SIZE)`. You may want to factor it out in a macro as it is extensively used.